finagle-core/src/test/scala/com/twitter/finagle/loadbalancer/StickyBalancerTest.scala

```diff
@@ -0,0 +1,150 @@
+package com.twitter.finagle.loadbalancer
+
+import com.twitter.app.App
+import com.twitter.conversions.time._
+import com.twitter.finagle._
+import com.twitter.finagle.stats.{StatsReceiver, NullStatsReceiver, InMemoryStatsReceiver}
+import com.twitter.util.{Function => _, _}
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+import scala.collection.mutable
+
+private[loadbalancer] trait StickySuite {
+  // number of servers
+  val N: Int = 100
+  // number of reqs
+  val R: Int = 100000
+
+  trait SServiceFactory extends ServiceFactory[Unit, Int]
+
+  val noBrokers = new NoBrokersAvailableException
+
+  def newBal(
+    fs: Var[Traversable[SServiceFactory]],
+    sr: StatsReceiver = NullStatsReceiver
+  ): ServiceFactory[Unit, Int] = new StickyBalancer(
+    Activity(fs.map(Activity.Ok(_))),
+    statsReceiver = sr,
+    emptyException = noBrokers,
+    maxEffort = 1
+  )
+}
+
+@RunWith(classOf[JUnitRunner])
+class StickyBalancerTest extends FunSuite with StickySuite {
```

Comment:
please add a test to check that it behaves correctly if traffic distributor hands you the same nodes as everyone else in the same order.